### PR TITLE
Add intel oneapi windows build to CI CMake

### PR DESCRIPTION
**Pull request by byrnHDF**
Replaces #3738

### CODE REVIEW

File: fortran/src/CMakeLists.txt [hunk 1]
```diff
@@ -13,8 +13,10 @@ if (WIN32)
       if (NOT H5_FORTRAN_C_LONG_DOUBLE_IS_UNIQUE)
         set (H5_DBL_NOT_UNIQUE_EXP ";")
       endif ()
+      set (CMAKE_NO_DEPRECATED_SYMBOLS 0)
       if (NOT H5_NO_DEPRECATED_SYMBOLS)
-        set (H5_NO_DEPRECATED_SYMBOLS ";")
+        set (CMAKE_NO_DEPRECATED_SYMBOLS 1)
```
Comment (mattjala) [inline fortran/src/CMakeLists.txt:19]: This seems to flip whether or not deprecated symbols are enabled - is this intentional?

Comment (byrnHDF) [reply fortran/src/CMakeLists.txt:19]: No that was not the intention, it was to only create a variable that could be used in the Fortran include file to create the correct define.

Comment (byrnHDF) [reply fortran/src/CMakeLists.txt:19]: See lines 93+ in H5config_f.inc.cmake

**Review by mattjala — APPROVED** [review]

File: fortran/src/CMakeLists.txt [hunk 2]
```diff
@@ -374,10 +384,6 @@ if (BUILD_SHARED_LIBS)
       PUBLIC ${HDF5_F90_C_LIBSH_TARGET}
       PRIVATE ${LINK_Fortran_LIBS} "$<$<BOOL:${HDF5_ENABLE_PARALLEL}>:${MPI_Fortran_LIBRARIES}>"
   )
-#  set_property(TARGET ${HDF5_F90_LIBSH_TARGET} APPEND PROPERTY LINK_FLAGS "$<$<STREQUAL:x${CMAKE_Fortran_SIMULATE_ID},xMSVC>:-SUBSYSTEM:CONSOLE>")
-#  set_property(TARGET ${HDF5_F90_LIBSH_TARGET} APPEND PROPERTY LINK_FLAGS "$<$<STREQUAL:x${CMAKE_Fortran_SIMULATE_ID},xMSVC>:${WIN_LINK_FLAGS}>")
-#  set_property(TARGET ${HDF5_F90_LIBSH_TARGET} APPEND PROPERTY LINK_FLAGS "$<$<STREQUAL:x${CMAKE_Fortran_SIMULATE_ID},xMSVC>:-DLL>")
-#  set_property(TARGET ${HDF5_F90_LIBSH_TARGET} APPEND PROPERTY LINK_FLAGS "$<$<STREQUAL:x${CMAKE_Fortran_SIMULATE_ID},xMSVC>:-DEF:${HDF5_F90_SRC_BINARY_DIR}/hdf5_fortrandll.def>")
   if(MSVC)
```
Comment (brtnfld) [inline fortran/src/CMakeLists.txt:387]: Is this still needed? It passes without it at least.

Comment (byrnHDF) [reply fortran/src/CMakeLists.txt:387]: I meant to delete the def part.

File: fortran/src/H5Pff.F90 [hunk 3]
```diff
@@ -5191,6 +5191,68 @@ END FUNCTION h5pget_fapl_mpio_c
...
+         IMPORT :: HID_T
+         IMPLICIT NONE
+         INTEGER(HID_T), INTENT(IN)  :: prp_id
+         INTEGER       , INTENT(OUT) :: data_xfer_mode
+       END FUNCTION h5pget_dxpl_mpio_c
+    END INTERFACE
+
+    hdferr = h5pget_dxpl_mpio_c(prp_id, data_xfer_mode)
+  END SUBROUTINE h5pget_dxpl_mpio_f
+
```
Comment (brtnfld) [inline fortran/src/H5Pff.F90]: I think this change is already in develop?

Comment (byrnHDF) [reply fortran/src/H5Pff.F90]: Moved the functions up as part of debugging - I can revert that

**Review by brtnfld — APPROVED** [review]
lgtm

**Pull request merged**